woocommerce/wc-admin
# Update Customer Effort Score Modal

This PR updates the placeholder modal with a new `Feedback` component containing [WordPress package components](https://developer.wordpress.org/block-editor/packages/packages-components/). 

The `RadioControl` component has been customized to display emoji buttons for the user to provide a score.

When a score of 1 or 2 is given, an optional comments field is provided.

An error message appears if the feedback is submitted without a score.

A basic small screen layout is in place. It is expected this will be refined further in later updates.

### Accessibility


### Screenshots

#### Small Screen
![Mobile](https://user-images.githubusercontent.com/9312929/97831124-1a1e3d80-1d0a-11eb-8bdb-20efcaeee458.png)

#### Desktop Screen
![Desktop](https://user-images.githubusercontent.com/9312929/97831235-5fdb0600-1d0a-11eb-8647-f59bd08517bd.png)

### Detailed test instructions:

- Delete any previous WP options for Customer Effort Score, e.g. `$ wp option delete woocommerce_ces_shown_for_actions`
- Update an existing product.
- On the next page, wait for the notification and press Give Feedback.

<img width="436" alt="Screen Shot 2020-11-18 at 2 13 50 pm" src="https://user-images.githubusercontent.com/9312929/99492380-6cb55600-29a8-11eb-93b2-12df1e819dfe.png">

- Confirm feedback modal appears
- Confirm clicking on send button validates score and displays error message
- Confirm hovering over emoji button displays hover state with blue background
- Confirm clicking emoji button displays active state with blue background
- Confirm comments field displays for the "Very Difficult" and "Somewhat Difficult" scores
- Resize window to small size and confirm modal is responsive.
- Close modal window
